test/e2e/upgrade/alert/alert.go

```diff
@@ -0,0 +1,167 @@
+package alert
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	helper "github.com/openshift/origin/test/extended/util/prometheus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	"k8s.io/kubernetes/test/e2e/upgrades"
+)
+
+const (
+	// Delay after upgrade is complete before checking for critical alerts
+	alertCheckSleepMinutes = 5
+	alertCheckSleep        = alertCheckSleepMinutes * time.Minute
+
+	// Previous period in which to check for critical alerts
+	alertPeriodCheckMinutes = 1
+
+	// Number of attempts if prometheus query fails
+	maxPrometheusQueryAttempts = 5
+
+	// Deley between each query attempt
+	prometheusQueryRetrySleep = 10 * time.Second
+)
+
+// UpgradeTest runs post-upgrade after alertCheckSleep delay and tests if any critical alerts are firing.
+type UpgradeTest struct {
+	url         string
+	bearerToken string
+	oc          *exutil.CLI
+}
+
+func (UpgradeTest) Name() string { return "check-for-critical-alerts" }
+func (UpgradeTest) DisplayName() string {
+	return "Check if critical alerts are firing after upgrade success"
+}
+
+// Setup creates parameters to query Prometheus
+func (t *UpgradeTest) Setup(f *framework.Framework) {
+	g.By("Setting up post-upgrade alert test")
+
+	// Must use version that's can run within Ginkgo It
+	oc := exutil.NewCLIWithoutNamespaceInside("prometheus")
+
+	var ok bool
+	url, bearerToken, ok := helper.LocatePrometheus(oc)
```

Comment:
I would reduce this to "ExpectPrometheus" and have that be a helper, it shouldn't require anything except a framework.